### PR TITLE
Remove `x-hacker` header.

### PR DESCRIPTION
**Pull request by wvffle**
What does this implement/fix? Explain your changes.
---------------------------------------------------
It lowers the amount of data sent.

Does this close any currently open issues?
------------------------------------------
No.

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
 It is literally a litter in the code and in the requests.  There is absolutely no point in sending that header.

Where has this been tested?
---------------------------
**Operating System:** Archlinux

**WordPress Version:** N/A

### CODE REVIEW

**Review by dzikoysk — APPROVED** [review]

**Comment by wvffle** [comment]
a screenshot for context:
image

**Comment by jasonbahl** [comment]
@wvffle The headers can be modified, including removing this header, using this filter here: https://github.com/wp-graphql/wp-graphql/blob/bb115be5ea8430cb3f36f68a6ce2939e72b8fe06/src/Router.php#L352

**Comment by jasonbahl** [comment]
@wvffle if you update the PR to make the header conditional to when debug mode is active, I'm cool with it.

File: src/Router.php [hunk 1]
```diff
@@ -330,7 +330,6 @@ protected static function get_response_headers() {
 			'Content-Type'                 => 'application/json ; charset=' . get_option( 'blog_charset' ),
 			'X-Robots-Tag'                 => 'noindex',
 			'X-Content-Type-Options'       => 'nosniff',
-			'X-hacker'                     => __( 'If you\'re reading this, you should visit github.com/wp-graphql and contribute!', 'wp-graphql' ),
 		];
```
Comment (jasonbahl) [inline src/Router.php:333]:
```suggestion
		];
		
		if ( true === WPGraphQL::debug() ) {
		  $headers[X-hacker] = __( 'If you\'re reading this, you should visit github.com/wp-graphql/wp-graphql and contribute!', 'wp-graphql' )
		}
```

**Review by jasonbahl — APPROVED** [review]

File: src/Router.php [hunk 2]
```diff
@@ -330,8 +330,11 @@ protected static function get_response_headers() {
...
+		
+		if ( true === WPGraphQL::debug() ) {
+		  $headers['X-hacker'] = __( 'If you\'re reading this, you should visit github.com/wp-graphql/wp-graphql and contribute!', 'wp-graphql' )
```
Comment (jasonbahl) [inline src/Router.php:336]:
```suggestion
		  $headers['X-hacker'] = __( 'If you\'re reading this, you should visit github.com/wp-graphql/wp-graphql and contribute!', 'wp-graphql' );
```

**Pull request merged**